### ISSUE TITLE
DTR signal issue with Arduino boards and in-circuit flashing

### ISSUE THREAD

**Issue by aremes** [nodemcu/nodemcu-flasher#8]
Hello!

I have run into a bit of an issue: I'm using the ESP module together with an Arduino (which, I am sure, a lot of other people are..) I figured I could hook up the Arduino to the ESP, then use software to pull GPIO0 low to put it in a mode in which flashing the ESP Firmware was possible. That didn't work for two reasons:

* The Arduino guys seem to use the DTR signal to reset the uC (so that whenever you start the SerialConsole from the IDE, the program would reset).. don't ask me why. Its fucking stupid. But that's the way it is...
* You guys seem to use the DTR signal to put an ESP (connected specifically to any UART chip like an FTDI) into flashing mode by connecting DTR to GPIO0. right?

Now. Please excuse me if i'm wrong about the latter part. If I am not wrong, then sorry: but thats fucking stupid too!  UART specification reserves use of that pin for specific purposes....

just because nobody (seriously NOBODY) uses flow control anymore, that doesnt mean you can use that signal for whatever you want.

Why not? Because the very fact that two different parties (Arduino & you guys) thought that the DTR signal was a convenient PC->embedded GPIO, those two things now seized to work well together.

Basically, when your software tries to flash the device (and pulls DTR low), it resets the Arduino making any "UART forwarding" and thus flashing of the device in-circuit impossible.

Please, tell me if i am wrong here...but the problem i just pointed out makes sense and matches what i observe at my end... so.. 

can you maybe make use of the DTR signal optional in the settings? That'd  be awesome. (and the RTS signal as well. Your code seems to be pulling that low too, for whatever reason)

I'd issue a pull request, but .. i'm not quite comfortable with the language, let alone that i have no idea how to compile that into working binaries. (seriously: give me a few hints on how to do that, and i'll issue a pull request ASAP!)
